pallets/anchors-v2/src/lib.rs

```diff
@@ -0,0 +1,287 @@
+// Copyright 2021 Centrifuge Foundation (centrifuge.io).
+// This file is part of Centrifuge chain project.
+
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::pallet_prelude::*;
+use frame_support::traits::ReservableCurrency;
+use frame_system::pallet_prelude::*;
+use scale_info::TypeInfo;
+use sp_runtime::{traits::AtLeast32BitUnsigned, FixedPointOperand};
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+pub mod weights;
+
+pub use pallet::*;
+pub use weights::*;
+
+#[derive(Encode, Decode, Clone, Eq, PartialEq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct Anchor<AccountId, DocumentId, DocumentVersion, Hash, Balance> {
+	account_id: AccountId,
+	document_id: DocumentId,
+	document_version: DocumentVersion,
+	hash: Hash,
+	deposit: Balance,
+}
+
+pub type AnchorOf<T> = Anchor<
+	<T as frame_system::Config>::AccountId,
+	<T as Config>::DocumentId,
+	<T as Config>::DocumentVersion,
+	<T as frame_system::Config>::Hash,
+	<T as Config>::Balance,
+>;
```

Comment:
NIT. Could we use `Anchor<T: Config>` instead? Much easier to refactor in the future

Reply:
It's what I did initially and got a warning:

```
warning: bounds on generic parameters are not enforced in type aliases
  --> pallets/anchors-v2/src/lib.rs:42:22
   |
42 | pub type AnchorOf<T: Config> =
   |                      ^^^^^^
```

Reply:
I meant in the struct:

```rust
pub struct Anchor<T: Config> {
	account_id: T::AccountId,
	document_id: T::DocumentId,
	document_version: T::DocumentVersion,
	hash: T::Hash,
	deposit: T::Balance,
}
```

Reply:
Updated, thank you!